scanner.cpp

```diff
@@ -0,0 +1,63 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <ctype.h>
+
+using namespace std;
+
+// Takes a character as an argument
+// Returns TRUE if character matches a self deliminating token.
+bool isSelfDeliminator(char c) {
+  char selfDeliminators[] = "+-*/<=(),:{;}";
+  for(int i = 0; i < 13; i++) {
```

Comment:
Great point; changed it to do this (and changed it to a C++ string to be easier to read).